src/components/common/decorators/watch.ts

```diff
@@ -34,7 +34,7 @@ export function watch(propName: string, options?: WatchOptions) {
 
         if (oldValue !== newValue) {
           if (!options?.waitUntilFirstUpdate || this.hasUpdated) {
-            this[name].call(this, oldValue, newValue);
+            this[name].call(this, newValue, oldValue);
```

Comment:
I think we should leave this as it was before since you may need only the old value, the new value could be retrieved from the property itself.